code/modules/power/singularity/narsie.dm

```diff
@@ -1,74 +1,193 @@
-/obj/singularity/narsie //Moving narsie to a child object of the singularity so it can be made to function differently. --NEO
-	name = "Nar'Sie's Avatar"
+#define NARSIE_CHANCE_TO_PICK_NEW_TARGET 5
+#define NARSIE_CONSUME_RANGE 12
+#define NARSIE_GRAV_PULL 10
+#define NARSIE_MESMERIZE_CHANCE 25
+#define NARSIE_MESMERIZE_EFFECT 60
+#define NARSIE_SINGULARITY_SIZE 12
+
+/// Nar'Sie, the God of the blood cultists
+/obj/narsie
+	name = "Nar'Sie"
 	desc = "Your mind begins to bubble and ooze as it tries to comprehend what it sees."
-	icon = 'icons/obj/narsie_small.dmi'
+	icon = 'icons/obj/narsie.dmi'
 	icon_state = "narsie"
-	pixel_x = -89
-	pixel_y = -85
+	anchored = TRUE
+	appearance_flags = LONG_GLIDE
 	density = FALSE
-	current_size = 9 //It moves/eats like a max-size singulo, aside from range. --NEO
-	contained = 0 //Are we going to move around?
-	dissipate = 0 //Do we lose energy over time?
-	move_self = 1 //Do we move on our own?
-	grav_pull = 5 //How many tiles out do we pull?
-	consume_range = 6 //How many tiles out do we eat
+	gender = FEMALE
+	layer = MASSIVE_OBJ_LAYER
+	light_color = COLOR_RED
 	light_power = 0.7
 	light_range = 15
-	light_color = COLOR_RED
-	gender = FEMALE
-
-/obj/singularity/narsie/large
-	name = "Nar'Sie"
-	icon = 'icons/obj/narsie.dmi'
-	// Pixel stuff centers Narsie.
+	light_range = 6
+	move_resist = INFINITY
+	obj_flags = CAN_BE_HIT | DANGEROUS_POSSESSION
 	pixel_x = -236
 	pixel_y = -256
-	current_size = 12
-	grav_pull = 10
-	consume_range = 12 //How many tiles out do we eat
-
-/obj/singularity/narsie/large/Initialize()
-	. = ..()
-	send_to_playing_players("<span class='narsie'>NAR'SIE HAS RISEN</span>")
-	sound_to_playing_players('sound/creatures/narsie_rises.ogg')
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	flags_1 = SUPERMATTER_IGNORES_1
 
-	var/area/A = get_area(src)
-	if(A)
-		var/mutable_appearance/alert_overlay = mutable_appearance('icons/effects/cult_effects.dmi', "ghostalertsie")
-		notify_ghosts("Nar'Sie has risen in \the [A.name]. Reach out to the Geometer to be given a new shell for your soul.", source = src, alert_overlay = alert_overlay, action=NOTIFY_ATTACK)
-	narsie_spawn_animation()
-	UnregisterSignal(src, COMSIG_ATOM_BSA_BEAM) //set up in /singularity/Initialize()
+	/// The singularity component to move around Nar'Sie
+	var/datum/component/singularity/singularity
```

Comment:
this is gross, this should use signals

Reply:
As in, signals that just set properties? Pretty sure that's in the realm of "signals as proc calls instead of GetComponent" that we don't want.

Reply:
tbh should just make a narsie subtype of singularity because the way narsie is tangled with the component is yuck.

Reply:
Not really sure what to say other than I disagree :shrug:

A subtype is weird, they don't share a lot of behavior and previously it was a bunch of weird overrides.

Signals are weird--we've been told that using signals as procs is bad and that we should use GetComponent instead. What is currently being used is essentially the same as that, except cached.

Reply:
If you're keeping reference to component like this, it should probably handle removal/deletion of the component. From what i see right now that's not the case for either singulo or narsie.